### ISSUE TITLE
jupyter run-button: why api only in project?

### ISSUE THREAD

**Issue by haraldschilly** [sagemathinc/cocalc#6693]
while doing some dev around the jupyter run button in markdown, it didn't recognize the kernel info in a code block, despite that I was able to run it. It uses a instance kernel in my project, right?

In any case, right [here in this line](https://github.com/sagemathinc/cocalc/blob/50ee0fcbc0ed5bf811e813635383f277e929c6dd/src/packages/frontend/components/run-button/index.tsx#L130) it checks for `!jupyterApiEnabled` and my settings was false. Since I'm in a project, this shouldn't matter, right? I was able to run the kernel, though.

**Comment by williamstein** [comment]
>  Since I'm in a project, this shouldn't matter, right? 

Yes, this is a bug.  The original implementation had all the code in a project running in the global sandbox, and that code was then correct.  I realized it would be much more valuable to have the code run locally in the project, and rewrote it, but must have missed a case.  thanks for finding this.

**Comment by williamstein** [comment]
I'll fix this though, since it's subtle, since the same code gets run in a project and on the share server.

**Comment by haraldschilly** [comment]
ok, thx. After I wrote this I was sure this is a bug. I didn't check further, i.e. if there are other similar tests that need to be adjusted.
